tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -131,31 +144,134 @@ func (a *AwsProvider) CreateWindowsVM() (rerr error) {
 	instance, err := a.createInstance(a.imageID, a.instanceType, a.sshKey, networkInterface, workerIAM, userDataWinrm)
 
 	if err != nil {
-		return err
+		return nil, err
 	}
 	instanceID := *instance.InstanceId
 
 	// Wait until instance is running and associate a unique name tag to the created instance.
 	err = a.waitUntilInstanceRunning(instanceID)
 	if err != nil {
-		return fmt.Errorf("failed to wait till instance is running, %v", err)
+		return nil, fmt.Errorf("failed to wait till instance is running, %v", err)
 	}
 	_, err = a.createInstanceNameTag(instance, infraID)
 	if err != nil {
 		log.V(0).Info(fmt.Sprintf("failed to assign name for instance: %s, %v", instanceID, err))
 	}
 
-	err = resource.AppendInstallerInfo([]string{instanceID}, []string{}, a.resourceTrackerDir)
+	// Get the decrypted password
+	decryptedPassword, err := a.GetPassword(instanceID)
+	if err != nil {
+		return nil, err
+	}
+	// Get the public IP
+	publicIPAddress, err := a.GetPublicIP(instanceID)
 	if err != nil {
-		return fmt.Errorf("failed to record instance ID to file at '%s',instance will not be able to be deleted, "+
+		return nil, err
+	}
+	// Build new credentials structure to be used by other actors
+	credentials = types.NewCredentials(instanceID, publicIPAddress, decryptedPassword)
+	err = resource.AppendInstallerInfo([]string{instanceID}, []string{}, a.resourceTrackerDir)
+	if err != nil || credentials == nil {
+		return nil, fmt.Errorf("failed to record instance ID to file at '%s',instance will not be able to be deleted, "+
 			"%v", a.resourceTrackerDir, err)
 	}
 
 	// Output commandline message to help RDP into the created instance.
-	log.V(0).Info(fmt.Sprintf("Successfully created windows instance: %s, please RDP into the Windows instance.",
-		instanceID))
+	log.V(0).Info(fmt.Sprintf("Successfully created windows instance: %s, "+
```

Comment:
Please change to `log.Info()`. Less refactoring for the future.

Reply:
Done